### PR TITLE
Update to DCMTK 3.6.4

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

**Comment by jcfr** [comment]
@pieper @fedorov This update to the latest release of DCMTK 3.6.4 

Before integrating, I will let you test.

**Comment by jcfr** [comment]
I also suggest we tag the repo before integrating this.

File: README.md [hunk 1]
```diff
@@ -98,6 +98,8 @@ Add / Remove applications
 Re-configuring the project setting any of these options allows to customize the
 list of applications bundled in the `dcmjs.js` module:
 
+* [cda2dcm](http://support.dcmtk.org/docs/cda2dcm.html): `-DCMTK_APP_CDA2DCM:BOOL=[ON|OFF]`
+* [dcm2json](http://support.dcmtk.org/docs/dcm2json.html): `-DCMTK_APP_DCM2JSON:BOOL=[ON|OFF]`
```
Comment (fedorov) [inline README.md]: any concerns also adding `dcm2xml`, `dsr2xml`, `dsr2html` and `dsrdump`?

Comment (fedorov) [reply README.md]: Never mind - sorry for the confusion!

**Comment by jcfr** [comment]
Options are available, will enable them

**Comment by fedorov** [comment]
Not so simple.... I configured with `-DDCMJS_APP_dsr2xml`, no errors. No build errors either. But the app is not discovered:
```
$ node dcmjs.js dsr2xml                                                                                                        
dsr2xml not available !
```

Maybe I need to build from scratch. Will look into it later.

But in any case the flags in README are incorrect and should be fixed. I can make a PR later, if you like.

**Comment by jcfr** [comment]
It turns out that option will be ignored, try to modify the option in the file: https://github.com/commontk/dcmjs/blob/master/CMake/Externals/External_DCMTK.cmake#L47

**Comment by jcfr** [comment]
Master is updated to support passing of options (https://github.com/commontk/dcmjs/commit/46352f52511138405c5d9562b4b20989ee9d8cd2)

Options are expected to be of the form `BUILD_<appname>_APP:BOOL=ON|OFF`, see https://github.com/commontk/dcmjs#add--remove-applications for more details

**Comment by fedorov** [comment]
I confirm this is working (config with dsr2xml option ok, build ok, and I can run dsr2xml with `node dcmjs.js dsr2xml`)!

 👍

**Comment by jcfr** [comment]
Where you able to process some data with the CLI ? If yes, would mind sharing them here (if small) .. that way we could add a test

**Comment by fedorov** [comment]
I did not! And I was opposite from smart by deleting the build folder completely before starting the build of the master branch to test whether it works with 3.6.1 (I wanted to have a clean build, but didn't think to keep the other one) ... 👎 

Let me finish the master build, and I will test that. I can then rebuild 3.6.4 and test that too, if you think there could have been a regression.

Here is a dataset you can use for the test: https://github.com/QIICR/dcmqi/blob/master/data/sr-example/sr.dcm.

**Comment by fedorov** [comment]
This does not seem to work, unless I messed something up again:

```bash
$ node dcmjs.js dsr2xml                                                                                                                            
$dcmtk: dsr2xml v3.6.4 2018-11-29 $

dsr2xml: Convert DICOM SR file and data set to XML
usage: dsr2xml [options] dsrfile-in [xmlfile-out]

parameters:
  dsrfile-in                     DICOM SR input filename to be converted
  xmlfile-out                    XML output filename (default: stdout)
[...]
$ node dcmjs.js dsr2xml ~/github/dcmqi/data/sr-example/sr.dcm                                                       
F: dsr2xml: error (No such file or directory) reading file: /Users/fedorov/github/dcmqi/data/sr-example/sr.dcm
```

**Comment by fedorov** [comment]
Using "native" binary available locally:
```bash
$ dsr2xml ~/github/dcmqi/data/sr-example/sr.dcm                                         
<?xml version="1.0" encoding="UTF-8"?>
<report type="Comprehensive SR">
<sopclass uid="1.2.840.10008.5.1.4.1.1.88.33">ComprehensiveSRStorage</sopclass>
<modality>SR</modality>
[...]
```

**Comment by fedorov** [comment]
Thank you! I confirm the current branch is working!

**Comment by fedorov** [comment]
Shouldn't `nodejs` in the README be replaced with `node`?

**Comment by jcfr** [comment]
> Shouldn't nodejs in the README be replaced with node?

:+1:  Sounds good. I have both on my side but it is an older installation.

> I confirm the current branch is working!

Great. I will tag the current master with 3.6.1, and integrate this PR

**Comment by jcfr** [comment]
All set. 

see 
* https://github.com/commontk/dcmjs/releases/tag/dcmjs-3.6.1
* https://github.com/commontk/dcmjs#creating-a-release

**Pull request merged**